src/Eloquent/Model.php

```diff
@@ -319,21 +319,44 @@ protected function fromDecimal($value, $decimals)
         return new Decimal128($this->asDecimal($value, $decimals));
     }
 
+    /**
+     * Convert MongoDB objects to their string representations.
+     *
+     * This method converts MongoDB related objects (ObjectID, Binary, UTCDateTime)
+     * to their serialized representations, ensuring the values can be correctly
+     * serialized when the model is converted to JSON.
+     *
+     * @param mixed $value The value to convert
+     *
+     * @return mixed
+     */
+    protected function convertMongoObjects(mixed $value): mixed
+    {
+        if ($value instanceof ObjectID) {
+            $value = (string) $value;
+        } elseif ($value instanceof Binary) {
+            $value = (string) $value->getData();
+        } elseif ($value instanceof UTCDateTime) {
+            $value = $this->serializeDate($value->toDateTime());
+        } elseif (is_array($value)) {
+            foreach ($value as &$nestedValue) {
+                $nestedValue = $this->convertMongoObjects($nestedValue);
+            }
+        }
+
+        return $value;
+    }
+
     /** @inheritdoc */
     public function attributesToArray()
     {
         $attributes = parent::attributesToArray();
 
-        // Because the original Eloquent never returns objects, we convert
         // MongoDB related objects to a string representation. This kind
```

Comment:
Please revert this.
```suggestion
        // Because the original Eloquent never returns objects, we convert
        // MongoDB related objects to a string representation. This kind
```